### PR TITLE
Refactor AWS and Azure cloud scripts of server and client

### PR DESCRIPTION
**Pull request by IsaacYangSLA**
Fixes # .

Refactor and remove sections on AWS server, AWS client and Azure server and Azure client.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [x] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/lighter/impl/master_template.yml [hunk 1]
```diff
@@ -1863,8 +1662,8 @@ aws_start_cln_sh: |
     echo "Launching container with docker option ${DOCKER_OPTION}."
     ssh -f -i $KEY_FILE -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null ${DEST_SITE} \
     "docker run -d -v ${DEST_FOLDER}:${DEST_FOLDER} --network host ${DOCKER_OPTION} ${image_name} \
-    /bin/bash -c \"python -u -m nvflare.private.fed.app.client.client_train -m ${DEST_FOLDER} \
-    -s fed_client.json --set uid={~~SITE~~} secure_train=true config_folder=config org={~~ORG~~} \" " > /tmp/nvflare.log 2>&1 
+    /bin/bash -c \"python -u -m nvflare.private.fed.app.{~~type~~}.{~~type~~}_train -m ${DEST_FOLDER} \
+    -s fed_{~~type~~}.json --set {~~cln_uid~~} secure_train=true config_folder=config org={~~ORG~~} \" " > /tmp/nvflare.log 2>&1
```
Comment (SYangster) [inline nvflare/lighter/impl/master_template.yml:1666]: is this supposed to be `--set uid={~~cln_uid~~}` or do we not need the `uid=`?

File: nvflare/dashboard/application/blob.py [hunk 2]
```diff
@@ -157,20 +153,37 @@ def gen_server(key, first_server=True):
         _write(os.path.join(dest_dir, "server.key"), cert_pair.ser_pri_key, "b", exe=False)
         _write(os.path.join(dest_dir, "rootCA.pem"), project.root_cert, "b", exe=False)
         if not project.ha_mode:
+            azure_start_svr_header_sh = tplt.get_azure_start_svr_header_sh()
```
[the remark this answers is not in the document]
Comment (IsaacYangSLA) [reply nvflare/dashboard/application/blob.py:156]: There are two shell scripts.  One for AWS and one for Azure.  So they are already in separate files.

File: nvflare/lighter/impl/master_template.yml (hunk 1 above)
Comment (IsaacYangSLA) [reply nvflare/lighter/impl/master_template.yml:1666]: Originally, we have uid={~~cln_uid~~} and {~~cln_uid~~} will be replaced with the site name.  Now we replace {~~cln_uid~~} with uid=site1 (if the site name is site1).  This makes the results the same.

File: nvflare/dashboard/application/blob.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (IsaacYangSLA) [reply nvflare/dashboard/application/blob.py]: Good suggestion.  Updated in the latest commit.

File: nvflare/dashboard/application/blob.py [hunk 3]
```diff
@@ -160,17 +158,29 @@ def gen_server(key, first_server=True):
             _write(
                 os.path.join(dest_dir, get_csp_start_script_name("azure")),
                 utils.sh_replace(
-                    tplt.get_cloud_script_header() + get_csp_template("azure", "svr", template),
-                    {"server_name": entity.name, "ORG": ""},
+                    tplt.get_start_sh("azure", "server"),
```
[the remark this answers is not in the document]
Comment (IsaacYangSLA) [reply nvflare/dashboard/application/blob.py:164]: The last commit split the master template file into three so azure and aws portions are in separate files.

**Comment by IsaacYangSLA** [comment]
> can we further refactor the gen_xxx functions for example def gen_server(): gen_config() perpare_startup_dir() prepare_certificates() gen_start_stop_scripts() for csp in provide_cloud_provides: gen_csp_start_scripts(csp) gen_csp_stop_scripts(csp)
> 
> all aws related scripts go to aws_file all azure related scripts go to azure file include template replacements etc.

There are four combinations for CSP's AND participant types, azure+server, azure+client, aws+server, aws+client.  Therefore, one statement for one combination seems good enough and there seems no need to create a function for that.

File: nvflare/dashboard/application/blob.py [hunk 4]
```diff
@@ -378,16 +305,16 @@ def gen_user(key, id):
...
+        utils._write(
             os.path.join(dest_dir, "fl_admin.sh"),
             utils.sh_replace(template["fl_admin_sh"], replacement_dict),
             "t",
             exe=True,
         )
-        _write(os.path.join(dest_dir, "client.crt"), cert_pair.ser_cert, "b", exe=False)
-        _write(os.path.join(dest_dir, "client.key"), cert_pair.ser_pri_key, "b", exe=False)
-        _write(os.path.join(dest_dir, "rootCA.pem"), project.root_cert, "b", exe=False)
+        utils._write(os.path.join(dest_dir, "client.crt"), cert_pair.ser_cert, "b", exe=False)
```
[the remark this answers is not in the document]
Comment (IsaacYangSLA) [reply nvflare/dashboard/application/blob.py:315]: Yes.  These three steps are similar to server, client and admin.

File: nvflare/dashboard/application/blob.py [hunk 5]
```diff
@@ -254,48 +222,30 @@ def gen_client(key, id):
...
             "t",
             exe=True,
         )
-        _write(
+        utils._write(
             os.path.join(dest_dir, get_csp_start_script_name("aws")),
             utils.sh_replace(
-                tplt.get_cloud_script_header() + get_csp_template("aws", "cln", template),
-                {"SITE": entity.name, "ORG": entity.org},
+                tplt.get_start_sh("aws"),
```
[the remark this answers is not in the document]
Comment (IsaacYangSLA) [reply nvflare/dashboard/application/blob.py:247]: Azure (line 235) is before this aws.

File: nvflare/dashboard/application/blob.py [hunk 6]
```diff
@@ -121,56 +111,51 @@ def gen_server(key, first_server=True):
...
+                        "server_name": entity.name,
+                        "ORG": "",
+                    },
                 ),
                 "t",
                 exe=True,
             )
-            _write(
+            utils._write(
                 os.path.join(dest_dir, get_csp_start_script_name("aws")),
```
[the remark this answers is not in the document]
Comment (IsaacYangSLA) [reply nvflare/dashboard/application/blob.py]: Thanks.  It's refactored.

File: nvflare/lighter/tplt_utils.py [hunk 7]
```diff
@@ -16,6 +16,43 @@
 class Template:
     def __init__(self, template):
         self.template = template
+        self.supported_csps = ("azure", "aws")
 
     def get_cloud_script_header(self):
         return self.template.get("cloud_script_header")
+
+    def get_start_sh(self, csp, type="server"):
+        if csp == "aws":
```
[the remark this answers is not in the document]
Comment (IsaacYangSLA) [reply nvflare/lighter/tplt_utils.py]: Now a dynamic function is used to get those scripts.

**Pull request merged**